src/dbup-cli/ConfigurationHelper.cs

```diff
@@ -22,6 +22,14 @@ public static class ConfigurationHelper
                     return DeployChanges.To.SqlDatabase(connectionString)
                         .WithExecutionTimeout(timeout)
                         .Some<UpgradeEngineBuilder, Error>();
+                case Provider.AzureSql:
+                    var useAzureSqlIntegratedSecurity = !(connectionString.Contains("Password", StringComparison.InvariantCultureIgnoreCase) ||
+                                      connectionString.Contains("Integrated Security", StringComparison.InvariantCultureIgnoreCase) ||
+                                      connectionString.Contains("Trusted_Connection", StringComparison.InvariantCultureIgnoreCase));
+
+                    return DeployChanges.To.SqlDatabase(connectionString, null, useAzureSqlIntegratedSecurity)
```

Comment:
Sorry, if this is work in progress - does EnsureDb + DropDb also need to us this overload in the same way?

Reply:
DbUp does not provide these overloads so I had to use a trick to do this. It's not a very beautiful solution but it should work.

Reply:
It all works thanks, @drwatson1 - PR looks great.